common/buildcraft/factory/tile/TilePump.java

```diff
@@ -90,7 +90,7 @@ private void buildQueue() {
         isInfiniteWaterSource = false;
         Set<BlockPos> checked = new HashSet<>();
         List<BlockPos> nextPosesToCheck = new ArrayList<>();
-        for (BlockPos posToCheck = pos.down(); posToCheck.getY() > 0; posToCheck = posToCheck.down()) {
+        for (BlockPos posToCheck = pos.down(); posToCheck.getY() > CubicChunksChecker.worldMin; posToCheck = posToCheck.down()) {
```

Comment:
Assuming the start position is not above the world (which seems reasonable), you could use `world.isOutsideBuildHeight()`.